File: frontend/src/types/report.ts
export interface SalesReportDay {
  bucket: string;
  total_paise: number;
  bill_count: number;
}

export interface SalesReport {
  date_from: string;
  date_to: string;
  range_clamped: boolean;
  total_paise: number;
  bill_count: number;
  avg_bill_paise: number;
  daily: SalesReportDay[];
}

export interface GstSummary {
  date_from: string;
  date_to: string;
  range_clamped: boolean;
  subtotal_paise: number;
  discount_paise: number;
  cgst_paise: number;
  sgst_paise: number;
  total_paise: number;
  bill_count: number;
}

File: backend/app/schemas/report.py
from datetime import date, datetime

from pydantic import BaseModel


class SalesReportDayOut(BaseModel):
    bucket: datetime
    total_paise: int
    bill_count: int


class SalesReportOut(BaseModel):
    date_from: date
    date_to: date
    range_clamped: bool
    total_paise: int
    bill_count: int
    avg_bill_paise: int
    profit_paise: int
    daily: list[SalesReportDayOut]


class GstSummaryOut(BaseModel):
    date_from: date
    date_to: date
    range_clamped: bool
    subtotal_paise: int
    discount_paise: int
    cgst_paise: int
    sgst_paise: int
    total_paise: int
    bill_count: int
